### DISCUSSION TITLE
Static Download URLs for Latest Versions

### DISCUSSION THREAD

**Discussion by tcoliver** [microsoft/openjdk#32] [Q&A] [answered]
For automated deployments, having static links available to the latest stable version would be very much appreciated. Ideally the links would be pinned to major versions to limit the possibility of breaking changes. A example use case would be a MECM or Jamf script that downloads and installs the latest OpenJDK distribution though Software Center or Self Service. Any chance of static links for each OS arch being made available?

**Comment by brunoborges** [comment]
Should this URL be for major versions or for minor versions? For example, should we have a URL that only points to "MS Build of OpenJDK", and downloads the very latest, or should it be for version X (e.g. 11) and latest minor update (e.g. "11.0.10").

**Reply by tcoliver** [reply]
For my use case, just the latest for major versions (e.g., openJDK 8, openJDK 11, openJDK 15) would be enough, assuming the installer was able to upgrade minor versions seamlessly.

**Reply by brunoborges** [reply]
Would you be concerned if your builds (CICD) or applications stopped working because of a faulty minor update that got you by surprise?

**Reply by tcoliver** [reply]
We would take on that risk in using a "latest" download url and deploying the application with that in mind. By that I mean we would not be using such a method for "mission critical" applications. Instead, it would function as more of a convenience method of allowing employees on with limited privileges to install the software as needed. Servers hosting managment software for example would be excluded from installing untested versions.

**Comment by tcoliver** [comment]
One additional piece that would also be helpful to bring into this request is that a similar url with the matching md5 or similar hash also be available to verify the authenticity of the downloaded software to be verified.

**Comment by d3vel0per** [comment]
Adding to the above ideas:
1) Additional endpoints that return the latest GA, LTS and EA releases will be helpful as well
2) Static url pointing to the license file

The above shall be useful in automating deployments

**Reply by d3vel0per** [reply]
Ok. Got the url for license file: https://raw.githubusercontent.com/microsoft/openjdk/main/LICENSE
Presumably it would remain the same in foreseeable future

**Comment by adotcoop** [comment]
Amazon already provide this for Corretto, so something along those lines would be welcome - see https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/downloads-list.html

A static link redirecting to the latest version would allow for inclusion in projects such as https://github.com/aaronparker/evergreen

Alternatively publishing the binaries to github and tagging with the latest release could also help- for example, community builds of OpenJDK do this and means the binaries can be retrieved via the following

https://api.github.com/repos/ojdkbuild/ojdkbuild/releases/latest

**Reply by jaknz** [reply]
Agreed; if you were to publish release info on Github, it would be simple to query the release API to find out the latest version. E.g. something like this:

`curl -Ls https://api.github.com/repos/microsoft/openjdk/releases/latest`

**Comment by brunoborges** [comment, marked as answer]
Available at: https://learn.microsoft.com/en-gb/java/openjdk/download-major-urls
